### ISSUE TITLE
Avoid requiring VariablePointers capability if possible

### ISSUE THREAD

**Issue by kvark** [EmbarkStudios/rust-gpu#593]
Looking at https://github.com/gfx-rs/naga/issues/713, I noticed that the shader requires "VariablePointers" capability.
Fully supporting this capability is significant amount of work for us, and I think we can get pretty far (in making Naga useful) before we start supporting it. The particular shader, for example, validates just fine if this capability is not set.

Note: same can be done for "VulkanMemoryModel" capability and the extension.

**Comment by XAMPPRocky** [comment]
Thank you for your issue! Closing this as duplicate of #530 and will be fixed by #591
